### PR TITLE
Add some completer metrics

### PR DESCRIPTION
**Pull request by nick-drozd**
These are some completer metrics that will be useful in measuring proposed fixes for the flatline bug. There will be more.

### CODE REVIEW

File: validator/sawtooth_validator/journal/completer.py [hunk 1]
```diff
@@ -87,8 +88,13 @@ def __init__(self,
             self._unsatisfied_dependency_count = CounterWrapper(
                 metrics_registry.counter(
                     'completer_unsatisfied_dependency_count'))
+            self._seen_txns_length = GaugeWrapper(
+                metrics_registry.gauge(
+                    'completer_seen_txns_length'))
         else:
             self._unsatisfied_dependency_count = CounterWrapper()
+            self._seen_txns_length = GaugeWrapper()
+            self._incomplete_blocks_length = GaugeWrapper()
```
Comment (aludvik) [inline validator/sawtooth_validator/journal/completer.py]: I think this line belongs in another commit

File: validator/sawtooth_validator/concurrent/threadpool.py [hunk 2]
```diff
@@ -41,13 +42,26 @@ def __init__(self, max_workers=None, name='', trace=None):
...
-            # This is the same default as ThreadPoolExecutor, but we want to
-            # know how many workers there are for logging
-            self._max_workers = multiprocessing.cpu_count() * 5
         super().__init__(max_workers)
 
+        if metrics_registry:
+            # Tracks how long tasks are queued
+            self._time_in_queue_gauge = GaugeWrapper(
+                metrics_registry.gauge(
+                    'threadpool.time_in_queue'))
```
Comment (aludvik) [inline validator/sawtooth_validator/concurrent/threadpool.py]: This needs the threadpool name prefixed to it, or else all of the threadpools will be aggregated into one metric.

File: validator/sawtooth_validator/concurrent/threadpool.py [hunk 3]
```diff
@@ -41,13 +42,26 @@ def __init__(self, max_workers=None, name='', trace=None):
...
+            # Tracks how many workers are already in use
+            self._workers_already_in_use_gauge = GaugeWrapper(
+                metrics_registry.gauge(
+                    'threadpool.workers_already_in_use'))
+            # Tracks how long tasks take to run
+            self._task_run_time_gauge = GaugeWrapper(
+                metrics_registry.gauge(
+                    'threadpool.task_run_time'))
+        else:
+            self._time_in_queue_gauge = GaugeWrapper()
```
Comment (aludvik) [inline validator/sawtooth_validator/concurrent/threadpool.py]: What was the reason for not using the pyformance Timer type for this?

File: validator/sawtooth_validator/concurrent/threadpool.py [hunk 4]
```diff
@@ -88,31 +94,23 @@ def wrapper():
...
-            # pylint: disable=broad-except
-            except Exception:
-                LOGGER.exception(
-                    '(%s) Unhandled exception during execution of task %s',
-                    self._name,
-                    task_details)
-
-            end_time = time.time()
-            run_time = (end_time - start_time) * 1000.0
+            with self._task_run_timer.time():
```
Comment (peterschwarz) [inline validator/sawtooth_validator/concurrent/threadpool.py]: This needs to be a timer per thread in the pool.  This measurement won't make sense otherwise.
